miio/cooker.py

```diff
@@ -0,0 +1,706 @@
+            "remaining=%s, " \
+            "cooking_delayed=%s, " \
+            "cooking_temperature=%s, " \
+            "settings=%s, " \
+            "interaction_timeouts=%s, " \
+            "hardware_version=%s, " \
+            "firmware_version=%s, " \
+            "favorite=%s, " \
+            "custom=%s>" % \
+            (self.mode,
+             self.menu,
+             self.stage,
+             self.temperature,
+             self.start_time,
+             self.remaining,
+             self.cooking_delayed,
+             self.duration,
+             self.settings,
+             self.interaction_timeouts,
+             self.hardware_version,
+             self.firmware_version,
+             self.favorite,
+             self.custom)
+        return s
+
+
+class Cooker(Device):
+    """Main class representing the cooker."""
+    @command(
+        default_output=format_output(
+            "",
+            "Mode: {result.mode}\n"
+            "Menu: {result.menu}\n"
+            "Stage: {result.stage}\n"
+            "Temperature: {result.temperature}\n"
+            "Start time: {result.start_time}\n"
+            "Remaining: {result.remaining}\n"
+            "Cooking delayed: {result.cooking_delayed}\n"
+            "Duration: {result.duration}\n"
+            "Settings: {result.settings}\n"
+            "Interaction timeouts: {result.interaction_timeouts}\n"
+            "Hardware version: {result.hardware_version}\n"
+            "Firmware version: {result.firmware_version}\n"
+            "Favorite: {result.favorite}\n"
+            "Custom: {result.custom}\n"
+        )
+    )
+    def status(self) -> CookerStatus:
+        """Retrieve properties."""
+        properties = ['func', 'menu', 'stage', 'temp', 't_func', 't_precook',
+                      't_cook', 'setting', 'delay', 'version']
+        values = self.send("get_prop", properties)
+
+        properties_count = len(properties)
+        values_count = len(values)
+        if properties_count != values_count:
+            _LOGGER.debug(
+                "Count (%s) of requested properties does not match the "
+                "count (%s) of received values.",
+                properties_count, values_count)
+
+        return CookerStatus(
+            defaultdict(lambda: None, zip(properties, values)))
+
+    @command(
+        click.argument("profile", type=str),
+        default_output=format_output("Cooking profile started"),
+    )
+    def start(self, profile: str):
+        """Start cooking a profile."""
+        if not self._validate_profile(profile):
+            raise CookerException("Invalid cooking profile: %s" % profile)
+
+        self.send('set_start', [profile])
+
+    @command(
+        default_output=format_output("Cooking stopped"),
+    )
+    def stop(self):
+        """Stop cooking."""
+        self.send('set_func', ['end02'])
+
+    @command(
+        default_output=format_output("Cooking stopped"),
+    )
+    def stop_outdated_firmware(self):
```

Comment:
Is it possible to check if outdated stop is necessary, and just expose the `stop` command?

Reply:
It's possible but ugly. I've to compare a lot of firmware versions and model types. It was to complicated to implement at the first step.

Reply:
Ok, then let it be like that for now.